who_is_active.sql

```diff
@@ -3679,7 +3679,50 @@ BEGIN;
 			@statement_start_offset INT,
 			@statement_end_offset INT,
 			@start_time DATETIME,
-			@database_name sysname;
+			@database_name SYSNAME,
+			@server_name SYSNAME = @@SERVERNAME,
+			@skip_databases_sql NVARCHAR(MAX) = N'
+				SELECT
+					d.name
+				FROM sys.databases AS d
+				INNER JOIN sys.availability_replicas AS r
+					ON d.replica_id = r.replica_id
+				WHERE NOT EXISTS
+					  (
+						  SELECT
+							  1/0
+						  FROM sys.dm_hadr_availability_group_states AS s
+						  WHERE s.primary_replica = r.replica_server_name
+					  )
+				AND r.secondary_role_allow_connections_desc = ''READ_ONLY''

```

Comment:
Are you sure about this logic? Reading the docs -- I don't have access to any AG servers at the moment -- I think we need to look for the string 'NO' here, not 'READ_ONLY'.

Also, I don't understand the necessity of the check on `sys.dm_hadr_availability_group_states`. If the current server is the primary, shouldn't that imply that the database will be readable?

Reply:
This is the logic we use in the first responder kit to avoid the errors we were hitting there when AGs were in use. I also don't currently have access to an AG for testing, and I can't recall all of the details.

Reply:
Okay, I think the exists check kind of makes sense now that I look at the DMVs in more detail. It looks like `availability_replicas` is supposed to have the same values no matter what server you query from. So you need to go to the other view to figure out whether or how those values apply to you. Maybe? These views are really poorly designed, and this reminds me why I hate dealing with SQL Server HA stuff. Anyway, I'm going to ask the original reporter to come back with some more info before we do anything else here.